src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -838,3 +839,127 @@ def get_connected_passed_kwargs(prefix):
         )
 
     return init_kwargs
+
+
+def _maybe_raise_warning_for_variant_checkpoint_format(
```

Comment:
I think it's better off as an in-line check because it is important for a developer to know right off the bat.

Reply:
agree here, not need to make a function for very simple logic like this

Reply:
reverted these changes